vm.go

```diff
@@ -617,17 +618,25 @@ func (vm *vm) run() {
 				ctx = context.Background()
 			}
 
-			if waitErr := vm.r.limiter.WaitN(ctx, vm.r.limiterTicksLeft); waitErr != nil {
-				if vm.r.vm.ctx == nil {
-					panic(waitErr)
-				}
-				if ctxErr := vm.r.vm.ctx.Err(); ctxErr != nil {
-					panic(ctxErr)
-				}
-				if strings.Contains(waitErr.Error(), "would exceed") {
-					panic(context.DeadlineExceeded)
+			now := time.Now()
+			r := vm.r.limiter.ReserveN(now, vm.r.limiterTicksLeft)
+			if !r.OK() {
+				panic("failed to make reservation")
```

Comment:
```suggestion
				panic(context.DeadlineExceeded)
```

This will keep the same behavior as the `if strings.Contains(waitErr.Error(), "would exceed") {` check. That being said, I don't believe this can ever happen at the moment.